src/test/java/org/takes/rs/RsWithTypeTest.java

```diff
@@ -222,5 +300,70 @@ public void replacesTypeWithText() throws Exception {
                 )
             )
         );
+        MatcherAssert.assertThat(
+            new RsPrint(
+                new RsWithType.Text(
+                    new RsWithType(
+                        new RsEmpty(), TYPE_HTML
+                    ),
+                    StandardCharsets.ISO_8859_1
+                )
+            ).print(),
+            Matchers.equalTo(
+                Joiner.on(CRLF).join(
+                    HTTP_OK,
+                    String.format(
+                        CONTENT_TYPE_WITH_CHARSET,
+                        TYPE_TEXT,
+                        StandardCharsets.ISO_8859_1
+                    ),
+                    "",
+                    ""
+                )
+            )
+        );
+    }
+
+    /**
+     * Ensures that the charset set is properly added when explicitly
+     * specified and absent when not specified.
+     * @throws java.lang.Exception If a problem occurs.
+     */
+    @Test
+    public void withCharset() throws Exception {
+        MatcherAssert.assertThat(
+            new RsPrint(
+                new RsWithType(
+                    new RsEmpty(), TYPE_TEXT, StandardCharsets.ISO_8859_1
+                )
+            ).print(),
+            Matchers.equalTo(
+                Joiner.on(CRLF).join(
+                    HTTP_OK,
+                    String.format(
+                        CONTENT_TYPE_WITH_CHARSET,
+                        TYPE_TEXT,
+                        StandardCharsets.ISO_8859_1
+                    ),
+                    "",
+                    ""
```

Comment:
@essobedo please put as much as possible to one line: http://www.yegor256.com/2014/04/27/typical-mistakes-in-java-code.html#indentation